contracts/Remittance.sol

```diff
@@ -1,16 +1,134 @@
 
+    struct RemittanceInstance {
+        address funder;
+        uint256 fundsOwed;
+        uint256 expiryDate;
+    }
+
+    event RemittanceFundsCreated(
+        bytes32 indexed hash,
+        address indexed funder,
+        address indexed broker,
+        uint256 amount,
+        uint256 expiryDate
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        bytes32 indexed password,
+        address indexed broker,
+        uint256 amount
+    );
+
+    event RemittanceFundsReclaimed(
+        bytes32 indexed hash,
+        address indexed funder,
+        uint256 amount
+    );
+
+    // The funder can create a remittance
+    // Hash is a sha256 hash of the broker address + recipient password
+    function create(bytes32 passwordBrokerHash, address broker, uint256 expiryDate)
+    external
+    payable
+    whenNotPaused
+    returns(bool success)
+    {
+        require(broker != msg.sender, "Caller cannot be the broker");
+        require(msg.value > 0, "Amount must be greater than 0");
+        require(broker != address(0), "Address cannot be zero");
+
+        // A 15 min margin has been added to allow for any lag
+        require(expiryDate >= block.timestamp + (24 * 60 * 60) - 1500, "Expiry less than 24h ahead");
+        checkEmptyHash(passwordBrokerHash);
+
+        RemittanceInstance storage remittanceInstance = remittances[passwordBrokerHash];
+
+        // .funder must be empty otherwise the key is already in use
+        require(remittanceInstance.funder == address(0), "Remittance already exists");
+
+        remittanceInstance.funder = msg.sender;
+        remittanceInstance.fundsOwed = msg.value;
+        remittanceInstance.expiryDate = expiryDate;
+
+        emit RemittanceFundsCreated(passwordBrokerHash, msg.sender, broker, msg.value, expiryDate);
+
+        return true;
+    }
+
+    // The broker sends the recipient's password to release the funds
+    function release(bytes32 password) external whenNotPaused returns(bool success) {
+
+        bytes32 rehash = hashPasswordBroker(password, address(msg.sender));
+
+        //Retrieve remittance
+        RemittanceInstance storage remittanceInstance = remittances[rehash];
+
+        uint256 fundsOwed = remittanceInstance.fundsOwed;
+
+        require(fundsOwed > 0, "No funds available");
+        require(block.timestamp < remittanceInstance.expiryDate, "Remittance has expired");
+
+        // Indicate remittance has been collected
+        remittanceInstance.fundsOwed = 0;
+        remittanceInstance.expiryDate = 0;
+
+        emit RemittanceFundsReleased(rehash, password, msg.sender, fundsOwed);
+
+        (success, ) = msg.sender.call{value: fundsOwed}("");
+        require(success, "Transfer failed");
+    }
+
+    // Funder can retrieve funds if not claimed by the expiry date
+    function reclaim(bytes32 passwordBrokerHash) external whenNotPaused returns(bool success) {
+
+        checkEmptyHash(passwordBrokerHash);
```

Comment:
I doubt it is necessary here. It costs gas for those who did not make a mistake, and for those who made this mistake, it would be caught lower on line 105. So let's make it cheaper for the majority that does not make the mistake.

Reply:
Yes. I see. Probably not so important in the final step.